### PR TITLE
schemes: Some improvements/simplifications

### PR DESCRIPTION
**Pull request by xdustinface**
Avoid some element creation and copying for no reason, be a bit more descriptive and type correct for some variables, drop some redundant code and simplify + improve readability in some places. Just walk through the commits hierarchically for easier review. Le me know your thoughts or concerns :)

Based on #150.

### CODE REVIEW

File: src/schemes.cpp [hunk 1]
```diff
@@ -171,15 +155,15 @@ bool CoreMPL::AggregateVerify(
     const vector<uint8_t> &signature)
 {
     int n = pubkeys.size();
+    G2Element signatureElement = G2Element::FromByteVector(signature);
     if (n <= 0 || n != messages.size()) {
```
Comment (mariano54) [inline src/schemes.cpp]: It looks like we are only checking args if we go into this if statement. Why not always check, for consistency with the other call locations?

I think a better way to do it, is to not check in the caller location. And instead do this:

if (!VerifyAggregateSignatureArguments(n, messages.size(), signatureElement) {
    return False
}

File: src/schemes.cpp [hunk 2]
```diff
@@ -393,14 +393,15 @@ bool PopSchemeMPL::FastAggregateVerify(
     const vector<uint8_t> &message,
     const G2Element &signature)
 {
+    if (pubkeys.size() <= 0) return false;
```
Comment (mariano54) [inline src/schemes.cpp]: Actually the infinity element signature should validate if there are no messages or public keys. I think we forgot to handle this edge case for FastAggregateVerify

**Review by mariano54 — COMMENTED** [review]
These are good changes, thanks! I'd like to hear your thoughts on my two comments

File: src/schemes.cpp (hunk 2 above)
Comment (mariano54) [reply src/schemes.cpp]: We don't have to address this is this PR. But the behavior should be the same for all schemes when verifying with no keys/message I think

Comment (xdustinface) [reply src/schemes.cpp]: Hm, actually it shouldn't matter in this case if i got it right. You will always have a message here which means https://github.com/Chia-Network/bls-signatures/blob/5d5da2778916cf39e5908372a36fd20c14f88724/src/schemes.cpp#L39 will anyway return `BAD` aka `false` and it couldn't even run into the "allow infinity" case.

File: src/schemes.cpp (hunk 1 above)
Comment (xdustinface) [reply src/schemes.cpp]: Makes sense 👍 I just unified it in all places in 0de8f04e19c2ce03761c1bbd1b75f44a30049e19, thoughts?

**Comment by xdustinface** [comment]
> I'd like to hear your thoughts on my two comments

Replied :) and also added one more: 316fb91

File: src/schemes.cpp (hunk 2 above)
Comment (mariano54) [reply src/schemes.cpp]: Oh okay that makes sense. These methods take in one public key and one message so there is no need to check

**Comment by mariano54** [comment]
LGTM

**Review by mariano54 — APPROVED** [review]

**Comment by hoffmang9** [comment]
Will merge shortly

**Pull request merged**